File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
import math

def is_prime(num):
    for i in range (2, int(math.sqrt(num)) + 1):
    	if num % i == 0:
        	return False
    return True
    
    
def solution(n, k):
    result = 0
    
    if k == 10:
        num = str(n)  
    else:
        num = ""
        temp = n
        
        while temp > 0:
            temp, mod = divmod(temp, k)
            num += str(mod)
        
        num = num[::-1]
    
    # 0 단위로 슬라이스 해서 소수 판별해야 할 듯?
    prime = num.split("0")
    prime_list = [prime_num for prime_num in prime if prime_num and is_prime(int(prime_num))]
    
    for j in prime_list:
        if j != '1':
            result += 1
    
    return result

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
import java.util.*;

class Solution {
    public int solution(int n, int k) {
        String s = Long.toString(n, k);  // k진수 변환
        String[] parts = s.split("0+");  // 0 기준 분리
        int answer = 0;
        
        for (String part : parts) {
            long num = Long.parseLong(part);
            
            if (isPrime(num)) answer++;
        }
        
        return answer;
    }

    // 4) 소수 판별 (long 범위)
    private boolean isPrime(long num) {
        if (num < 2) return false; // 0, 1 모두 소수 아님

        for (long i = 2; i <= Math.sqrt(num); i++) {
            if (num % i == 0) return false;
        }
        return true;
    }
}
